requires.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2021 Canonical Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from charms import reactive
+
+
+class BindClientRequires(reactive.Endpoint):
```

Comment:
This class misses the `__init__` constructor, inheriting from `Endpoint`. The provider class does define the constructor. Is this intended?

OTOH, I think this method could benefit from a method that collects possible attributes shared over the relation (e.g. `hostname` and `port`). This approach may not immediately help the `prometheus-bind-exporter-operator` charm because it is not reactive. But if there's ever a reactive charm using this interface, it won't need to use `relation_get` directly.

Reply:
>The provider class does define the constructor. Is this intended?

It wasn't intended. Great catch. Thanks!

>But if there's ever a reactive charm using this interface, it won't need to use relation_get directly.

I've searched a method in the `Endpoint` object and didn't find a method for this. The equivalent of relation_set can be used in the `provides.py`  with `publish_raw`. Do you have an example of this?